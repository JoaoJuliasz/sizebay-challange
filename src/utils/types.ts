export type FilterType = 'done' | 'pending' | ''

export type Items = {
    value: string,
    status: 'done' | 'pending'
    itemIndex: number
}