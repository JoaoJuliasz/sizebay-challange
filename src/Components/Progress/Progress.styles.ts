import styled from 'styled-components'

export const ProgressContainer = styled.div`
    width: 80%;
    margin: auto;

    @media only screen and (max-width: 600px) {
        width: 90%;
    }
`