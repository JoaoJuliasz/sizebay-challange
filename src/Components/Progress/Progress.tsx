import { useContext } from 'react'

// Importing libs
// @ts-ignore
import { Progress } from 'antd';

import { Items } from '../../utils/types'
import ProgressConfetti from '../ProgressConfetti/ProgressConfetti';
import { ProgressContainer } from './Progress.styles';

type ShowProgressProps = {
    items: Items[],
}

const ShowProgress = ({ items }: ShowProgressProps) => {

    const fillProgress = () => {
        let progressCounter = 0
        let itemsLenght = items.length
        items.map(item => {
            if (item.status.includes('done')) {
                progressCounter++
            }
        })
        return items.length > 0 ? progressCounter * 100 / itemsLenght : 0
    }

    return (
        <ProgressContainer>
            <Progress className="progressBar" showInfo={false} percent={fillProgress()} strokeColor={fillProgress() === 0 ? '#f5f5f5' : '#68e291'} />
            <ProgressConfetti percent={fillProgress()} />
        </ProgressContainer>
    );
};

export default ShowProgress;