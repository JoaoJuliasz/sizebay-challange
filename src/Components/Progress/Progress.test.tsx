import {shallow} from 'enzyme'
import { Items } from '../../utils/types'
import Progress from './Progress'

describe('Progress', () => {

    const mockItems:Items[] = [
        {value: 'item 1', status: 'done', itemIndex: 0},
        {value: 'item 2', status: 'pending', itemIndex: 1},
    ]

    const progressWrapper = shallow(<Progress items={mockItems}/>)

    it('expect to render Progress component', () => {
        expect(progressWrapper).toMatchSnapshot()
    })

    it('expect to progress bar percent be 50%', () => {
        const progressBar = progressWrapper.find('.progressBar')
        expect(progressBar.prop('percent')).toBe(50)
    })
})