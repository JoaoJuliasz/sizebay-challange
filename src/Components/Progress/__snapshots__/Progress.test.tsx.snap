// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress expect to render Progress component 1`] = `ShallowWrapper {}`;
